crates/napi/src/minify.rs

```diff
@@ -118,10 +118,19 @@ fn patch_opts(opts: &mut JsMinifyOptions) {
         .collect(),
         ..Default::default()
     });
-    opts.mangle = BoolOrDataConfig::from_obj(MangleOptions {
-        reserved: vec!["AbortSignal".into()],
-        ..Default::default()
-    })
+
+    if opts.mangle.is_true() {
+        opts.mangle = BoolOrDataConfig::from_obj(MangleOptions {
+            reserved: vec!["AbortSignal".into()],
+            ..Default::default()
+        });
+    } else if opts.mangle.is_obj() {
+        let mangle = std::mem::take(&mut opts.mangle);
+        opts.mangle = mangle.map(|mut mangle_opts| {
+            mangle_opts.reserved.push("AbortSignal".into());
+            mangle_opts
+        });
+    }
```

Comment:
As this is not a hot codepath, I'd personally prefer something that minimizes branching/duplication like:

```suggestion
    if opts.mangle.is_true() {
        opts.mangle = BoolOrDataConfig::from_obj(MangleOptions::default());
    }
    let mangle = std::mem::take(&mut opts.mangle);
    opts.mangle = mangle.map(|mut mangle_opts| {
        mangle_opts.reserved.push("AbortSignal".into());
        mangle_opts
    });
```